dotnet/targets/Xamarin.Shared.Sdk.targets

```
@@ -530,6 +530,11 @@
 			<!-- Suppress trimmer warnings unless we're trimming all assemblies -->
 			<SuppressTrimAnalysisWarnings Condition="'$(SuppressTrimAnalysisWarnings)' == '' And '$(TrimMode)' == 'full'">false</SuppressTrimAnalysisWarnings>
 			<SuppressTrimAnalysisWarnings Condition="'$(SuppressTrimAnalysisWarnings)' == ''">true</SuppressTrimAnalysisWarnings>
+			<!--
+				In the case of NativeAOT builds, the trim warnings will be produced either by the ILLink roslyn analyzer in Debug builds
+				or by the ILC when publishing. We want to suppress the ILLink warnings in all build configurations to avoid duplicates.
+			-->
+			<SuppressTrimAnalysisWarnings Condition="'$(PublishAot)' == 'true'">true</SuppressTrimAnalysisWarnings>
```

I think this should be:
```suggestion
			<SuppressTrimAnalysisWarnings Condition="'$(PublishAot)' == 'true' And '$(_IsPublishing)' != 'true'">true</SuppressTrimAnalysisWarnings>
```
As we could have a project file with both `'$(TrimMode)' == 'full'` and '$(PublishAot)' == 'true'.
If that is the case and we do `dotnet build`, ILC will not run, so we want warnings from ILLink. 
When both  `PublishAot` and `_IsPublishing` are `true` aka `_UseNativeAot=true` we already suppress the warnings in props, so we got that covered.

FWIW, to avoid confusion, I think we should rename `_UseNativeAot` to something like `_PublishWithNativeAot` or something like that.

---

Changing this condition won't make any difference, but it might make it a bit more confusing and harder to follow the values of `SuppressTrimAnalysisWarnings`. _"Why shouldn't this be set to `true` when publishing?"_ I don't have a strong preference but I would lean towards keeping the condition simple.

I agree that `_UseNativeAot` is a bit confusing. It's a combination of `PublishAot` and `_IsPublishing` so I suggest `_IsPublishingUsingNativeAot` or `_IsPublishingWithNativeAot`.